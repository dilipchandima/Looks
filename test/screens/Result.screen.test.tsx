import React from 'react';

import {fireEvent, render} from '@testing-library/react-native';
import ResultScreen from 'screens/Result.screen';

const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe('Result Screen', () => {
  it('should match to snapshot', () => {
    const screen = render(<ResultScreen navigation={navigation} />);
    expect(screen).toMatchSnapshot();
  });

  it('should render the retake and use photo buttons', () => {
    const {queryByText} = render(<ResultScreen navigation={navigation} />);
    expect(queryByText('Retake')).toBeTruthy();
    expect(queryByText('Use photo')).toBeTruthy();
  });

  it('should call navigation when press the Retake button', () => {
    const {getByText} = render(<ResultScreen navigation={navigation} />);
    const button = getByText('Retake');
    fireEvent.press(button);

    expect(navigation.goBack).toHaveBeenCalledTimes(1);
  });
});
