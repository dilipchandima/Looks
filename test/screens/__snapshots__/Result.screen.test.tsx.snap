// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Screen should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
      "paddingTop": 100,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 1000,
          "width": 750,
        }
      }
    />
    <Text>
      result
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "opacity": 1,
            "paddingHorizontal": 20,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#c74a4e"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              Object {
                "payload": 4291250766,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="M19 12L5 12"
            />
            <RNSVGPath
              d="M12 19L5 12 12 5"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "marginLeft": 10,
            }
          }
        >
          Retake
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 50,
            "opacity": 1,
            "paddingHorizontal": 20,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="#1ca340"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              Object {
                "payload": 4280066880,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          >
            <RNSVGPath
              d="M20 6L9 17 4 12"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "marginLeft": 10,
            }
          }
        >
          Use photo
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
